src/osd/PG.h

```diff
@@ -94,7 +94,7 @@ struct PGRecoveryStats {
     // cppcheck-suppress unreachableCode
     per_state_info() : enter(0), exit(0), events(0) {}
   };
-  map<const char *,per_state_info> info;
+  map<string, per_state_info> info;
```

Comment:
@stiopaa1 even it looks suspicious, it is actually correct, i think. the `const char *` keys are the `NamedState::state_name`, and they live in .rodata section, so

1. they won't go away.
1. and they are unique. in other words, two difference instances of the same subclass of `NamedState` will share the same `state_name`:
```c++
Crashed c1, c2;
assert(c1.state_name == c2.state_name);
```
1. and it's fine to use the addresses of the strings as the keys for info. as we don't care about the order in which the `per_state_info` instances are arranged in info. what we need is actually an unordered_map here.